### PR TITLE
Use pread for all remote hdfs api read calls

### PR DESCRIPTION
**Pull request by yuzhu**
In addition, make remote the default configuration for hdfs ufs

### CODE REVIEW

File: core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 1]
```diff
@@ -675,7 +675,7 @@ public String toString() {
           .build();
   public static final PropertyKey UNDERFS_HDFS_REMOTE =
       new Builder(Name.UNDERFS_HDFS_REMOTE)
-          .setDefaultValue(false)
+          .setDefaultValue(true)
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/conf/PropertyKey.java:678]: Please update the description

Comment (yuzhu) [reply core/common/src/main/java/alluxio/conf/PropertyKey.java:678]: updated

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
```ufsJournal(alluxio.server.ft.journal.TriggeredCheckpointTest)  Time elapsed: 127.767 sec  <<< ERROR!
java.util.concurrent.TimeoutException: Timed out waiting for all nodes registered options: WaitForOptions{interval=200, timeout=20000} last value: false
	at alluxio.server.ft.journal.TriggeredCheckpointTest.validateCheckpointInClusterRestart(TriggeredCheckpointTest.java:124)
	at alluxio.server.ft.journal.TriggeredCheckpointTest.ufsJournal(TriggeredCheckpointTest.java:63)

ufsJournal(alluxio.server.ft.journal.TriggeredCheckpointTest)  Time elapsed: 127.596 sec  <<< ERROR!
java.util.concurrent.TimeoutException: Timed out waiting for all nodes registered options: WaitForOptions{interval=200, timeout=20000} last value: false
	at alluxio.server.ft.journal.TriggeredCheckpointTest.validateCheckpointInClusterRestart(TriggeredCheckpointTest.java:124)
	at alluxio.server.ft.journal.TriggeredCheckpointTest.ufsJournal(TriggeredCheckpointTest.java:63)


Results :

Tests in error: 
alluxio.server.ft.journal.TriggeredCheckpointTest.ufsJournal(alluxio.server.ft.journal.TriggeredCheckpointTest)
  Run 1: TriggeredCheckpointTest.ufsJournal:63->validateCheckpointInClusterRestart:124 » Timeout
  Run 2: TriggeredCheckpointTest.ufsJournal:63->validateCheckpointInClusterRestart:124 » Timeout```

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Pull request merged**